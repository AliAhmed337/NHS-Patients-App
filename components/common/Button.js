import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import { Ionicons } from '@expo/vector-icons';



const Button = (props) => {

    return (

        <TouchableOpacity style={[styles.buttonBody, {backgroundColor: props.backgroundColor},{borderRadius: props.borderRadius }, {width: props.width},{height:props.height}, {padding: props.padding}, {paddingBottom: props.paddingBottom}, {paddingLeft: props.paddingLeft}, {borderColor: props.borderColor}]} onPress = {props.onPress} >

            <View style={{flexDirection: 'row',alignItems: 'center',justifyContent:'space-between', flex: 1}}>
                <Ionicons paddingLeft={props.paddingLeft} name={props.name} size={props.size} color={props.color} style={[styles.buttonImg]}/>
                <Text style={[styles.buttonText,{color: props.color},{fontSize: props.fontSize}]}>{props.children}</Text>
                <View></View>
            </View>

        </TouchableOpacity>

    )

}

const styles = ({
    buttonBody: {


        alignItems: 'center',
        flexDirection: 'row',
        justifyContent: 'space-evenly',
        shadowColor: '#000',


    },
    buttonText: {
        fontWeight: '600',
    },
    buttonImg: {


    }


})
export default Button;