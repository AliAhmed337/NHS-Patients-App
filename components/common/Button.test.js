import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Testing common button component', () => {

	it('renders correcly', () => {
		const wrapper = shallow(
			<Button 
				backgroundColor="black"
				width="100%" 
				height="100%" 
				padding={10}
				name="md-checkbox"
				size={20}
				color="white"
				/>
			);
		wrapper.setProps({ 
				backgroundColor: 'black',
				width: '100%', 
				height: '100%', 
				padding: 10,
				name: 'md-checkbox',
				size: 20,
				color: 'white'
				});
		expect(wrapper).toMatchSnapshot();

	});	

});
