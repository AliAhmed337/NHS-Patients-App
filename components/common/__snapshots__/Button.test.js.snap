// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing common button component renders correcly 1`] = `ShallowWrapper {}`;
